src/Compilers/Core/Portable/Operations/ChildOperationList.cs

```diff
@@ -13,49 +13,46 @@
 
 namespace Microsoft.CodeAnalysis
 {
-    internal abstract partial class Operation : IOperation
+    /// <summary>
+    /// Implements a struct-based enumerable for <see cref="Operation"/> nodes, using a slot-based system that tracks
+    /// the current slot, and the current index in the slot if the current slot is an immutable array. This type is not hardened
+    /// to <code>default(Enumerable)</code>, and will null reference in these cases.
+    /// </summary>
+    [NonDefaultable]
+    public readonly struct ChildOperationEnumerable : IEnumerable<IOperation>
     {
-        /// <summary>
-        /// Implements a struct-based enumerable for <see cref="Operation"/> nodes, using a slot-based system that tracks
-        /// the current slot, and the current index in the slot if the current slot is an immutable array. This type is not hardened
-        /// to <code>default(Enumerable)</code>, and will null reference in these cases.
-        /// </summary>
-        [NonDefaultable]
-        internal readonly struct Enumerable : IEnumerable<IOperation>
-        {
-            private readonly Operation _operation;
+        private readonly Operation _operation;
 
-            internal Enumerable(Operation operation)
-            {
-                _operation = operation;
-            }
+        internal ChildOperationEnumerable(Operation operation)
+        {
+            _operation = operation;
+        }
 
-            public Enumerator GetEnumerator() => new Enumerator(_operation);
+        public Enumerator GetEnumerator() => new Enumerator(_operation);
 
-            public ImmutableArray<IOperation> ToImmutableArray()
+        public ImmutableArray<IOperation> ToImmutableArray()
+        {
+            switch (_operation)
             {
-                switch (_operation)
-                {
-                    case NoneOperation { Children: var children }:
-                        return children;
-                    case InvalidOperation { Children: var children }:
-                        return children;
-                    case var _ when !GetEnumerator().MoveNext():
-                        return ImmutableArray<IOperation>.Empty;
-                    default:
-                        var builder = ArrayBuilder<IOperation>.GetInstance();
-                        foreach (var child in this)
-                        {
-                            builder.Add(child);
-                        }
-                        return builder.ToImmutableAndFree();
-                }
+                case NoneOperation { Children: var children }:
+                    return children;
+                case InvalidOperation { Children: var children }:
+                    return children;
+                case var _ when !GetEnumerator().MoveNext():
+                    return ImmutableArray<IOperation>.Empty;
+                default:
+                    var builder = ArrayBuilder<IOperation>.GetInstance();
```

Comment:
can you seed with the count?  or is it unknown?

Reply:
This is entirely unchanged from existing implementation. But yes, count is unknown unless we enumerate the list.